src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -93,13 +97,51 @@ export const MultiSelect: ChakraComponent<
         ...rest
       } = props;
 
+      const [userClickedOutside, setUserClickedOutside] =
+        useState<boolean>(false);
+
       // Create a ref to hold a reference to the accordian button, enabling us
       // to programmatically focus it.
       const accordionButtonRef: React.RefObject<HTMLDivElement> =
         useRef<HTMLDivElement>();
+      const containerRef: React.RefObject<HTMLDivElement> =
+        useRef<HTMLDivElement>();
       const expandToggleButtonRef: React.RefObject<HTMLButtonElement> =
         useRef<HTMLButtonElement>();
 
+      const handleEvent = (e) => {
+        // Handles when user clicks outside of the container
+        if (e.type === "mousedown") {
+          if (
+            containerRef.current &&
+            !containerRef.current.contains(e.target)
+          ) {
+            setUserClickedOutside(true);
+          } else {
+            setUserClickedOutside(false);
+          }
+        }
+
+        // Handles when user tabs outside of the container
+        if (e.type === "blur") {
+          if (!e.currentTarget.contains(e.relatedTarget)) {
+            setUserClickedOutside(true);
+          } else {
+            setUserClickedOutside(false);
+          }
+        }
+      };
```

Comment:
Can you try breaking this out into a separate function so that the mousedown event triggers `handleEvent` and the onblur event on the div element triggers another function with this code block? This is mostly to separate the two concepts.

Reply:
Yup, shouldn't force it if it doesn't make sense